ocaml/typing/shape.mli

```diff
@@ -65,7 +65,7 @@ module Item : sig
 end
 
 type var = Ident.t
-type t = { uid: Uid.t option; desc: desc }
```

Comment:
The new hash field essentially has an invariant on it that is ensured by the smart constructors. That probably means that this type should be made `private` if it can be.